src/audio/module_adapter/module/volume/volume_generic.c

```diff
@@ -82,17 +85,20 @@ static void vol_s24_to_s24(struct processing_module *mod, struct input_stream_bu
 			x0 = x + j;
 			y0 = y + j;
 			vol = cd->volume[j];
+#if CONFIG_COMP_PEAK_VOL
```

Comment:
So this is getting quite complex with the conditional build time logic. I would duplicate functions that are different for peak vol and minimize the number of conditional peakvol items to simplify the logic.

Reply:
or maybe a separate file?

Reply:
ack, separate file may be easier.

Reply:
Since add two files like volume_generic_with_peak_vol.c seems a little strange, is it okay to do so as bellow?
#if CONFIG_COMP_PEAK_VOL
.../*functions with peak volume detection*/
#else
.../*functions without peak volume detection*/
#endif